src/models/CommonModel.ts

```diff
@@ -84,6 +84,23 @@ export class CommonModel extends CommonSchema<CommonModel> {
     return this.required.includes(propertyName);
   }
 
+  /**
+   * Adds a property to the model.
+   * If the property already exist the two are merged.
+   * 
+   * @param propertyName 
+   * @param propertyModel 
+   * @param schema schema to the corresponding property model
+   */
+  addProperty(propertyName: string, propertyModel: CommonModel, schema: Schema) {
+    if (this.properties === undefined) this.properties = {};
+    if (this.properties[`${propertyName}`] !== undefined) {
+      this.properties[`${propertyName}`] = CommonModel.mergeCommonModels(this.properties[`${propertyName}`], propertyModel, schema);
```

Comment:
Can not you replace 
```
[`${propertyName}`]
```
with
```
[propertyName]
```
?
I am not sure why are you parsing it to a string if it is already a string...

Reply:
It is because of `Generic Object Injection Sink` warning in eslint. If the library is used in a JS setting any value can technically be passed to the function, meaning that the transpiled JS version of the function is 
```
addProperty(propertyName, propertyModel, schema) {
```
which does not check that the input is a string format, and any input from the user is accepted.

Reply:
Well I do not belive it is a good way to make this.
It would be better to make a validation for user input with Joi or something similiar and throw an error to tell a user that something is wrong. This is a standard to make validation for user input for every API. 
Now if you use propertyName somewhere else you MUST duplicate parsing to string.
Btw. I would prefer to see something like `String(propertyName)`.

Reply:
Using Joi might be a bit overkill for this 🤔 We use `${}` pretty consistent, so let's stay that way with this. Not that I have any strong opinion about it so it is all about preference, but let's stay consistent.

But we should only parse it once, gonna adapt it.

Reply:
Never mind, it is not that simple apparently, need to think about how to handle security/detect-object-injection another way. 

related: https://stackoverflow.com/questions/44882542/why-is-it-bad-pratice-calling-an-array-index-with-a-variable/44882765#44882765

Reply:
@jonaslagoni 
> We use ${} pretty consistent

This is only for injecting string into other like ```
`Creating ${entites.length} entities`
``` instead of `'Creating ' + entites.length + ' entites'`. It was not intendent to do weak parsing. 
Also if you may have something else than string why did you tell compiler that there will always be a string?
I think that you should ensure that this string is actually a string or fix wrong type.

> Using Joi might be a bit overkill for this
Why?

Reply:
> Why?

Because the only reason we have this problem is the `Generic Object Injection Sink`. We don't require complex formats that would require such a library, at least I don't see the benefits, at least not now. If the functions start to be regularly used by external users then it could be a good idea to add it, but not with the current state of the library.

Feel free to create a feature request with the suggested changes, that outlines the benefits of having such a library 🙂

> It was not intendent to do weak parsing.

> In parser we use `String(propertyName)` notation in every such a place :)

Fair enough, adding an issue to refactor it across all files as we use it multiple places.

Reply:
https://github.com/asyncapi/generator-model-sdk/issues/196